stablehlo/dialect/VhloBytecode.cpp

```diff
@@ -663,39 +889,218 @@ void VhloBytecodeInterface::write(ScatterDimensionNumbersAttr attr,
 //===----------------------------------------------------------------------===//
 // TransposeAttr
 
-TransposeAttr VhloBytecodeInterface::readTransposeAttr(
+TransposeV1Attr VhloBytecodeInterface::readTransposeV1Attr(
     DialectBytecodeReader &reader) const {
   LOG_READ_CALL;
-  return hlo::bytecode::readEnumAttribute<TransposeAttr>(
+  return hlo::bytecode::readEnumAttribute<TransposeV1Attr>(
       reader, getContext(),
-      [](uint32_t val) { return symbolizeTranspose(val); });
+      [](uint32_t val) { return symbolizeTransposeV1(val); });
 }
 
-void VhloBytecodeInterface::write(TransposeAttr attr,
+void VhloBytecodeInterface::write(TransposeV1Attr attr,
                                   DialectBytecodeWriter &writer) const {
   writer.writeVarInt(vhlo_encoding::kTransposeAttr);
-  hlo::bytecode::writeEnumAttribute<Transpose>(attr, writer);
+  hlo::bytecode::writeEnumAttribute<TransposeV1>(attr, writer);
 }
 
 //===----------------------------------------------------------------------===//
 // TypeExtensionsAttr
 
-TypeExtensionsAttr VhloBytecodeInterface::readTypeExtensionsAttr(
+TypeExtensionsV1Attr VhloBytecodeInterface::readTypeExtensionsV1Attr(
     DialectBytecodeReader &reader) const {
   LOG_READ_CALL;
   llvm::SmallVector<int64_t> bounds;
   if (failed(reader.readSignedVarInts(bounds))) {
-    return TypeExtensionsAttr();
+    return TypeExtensionsV1Attr();
   }
-  return TypeExtensionsAttr::get(getContext(), bounds);
+  return TypeExtensionsV1Attr::get(getContext(), bounds);
 }
 
-void VhloBytecodeInterface::write(TypeExtensionsAttr attr,
+void VhloBytecodeInterface::write(TypeExtensionsV1Attr attr,
                                   DialectBytecodeWriter &writer) const {
   writer.writeVarInt(vhlo_encoding::kTypeExtensionsAttr);
   writer.writeSignedVarInts(attr.getBounds());
 }
 
+//===----------------------------------------------------------------------===//
+// Forked Attributes
+//===----------------------------------------------------------------------===//
+
+namespace {
+template <typename AttrOrType>
+void assertFromVhlo(AttrOrType val) {
```

Comment:
Since we aren't asserting in this function anymore, let's rename it to something else to avoid giving a wrong impression. E.g. `verifyFromVhlo` or `checkFromVhlo`.

Reply:
Changed to `verifyFromVhlo`. This will still be needed for bytecode read, a verifier on the VHLO ops should cover the VHLO write pieces.